drivers/serial/uart_ns16550.c

```diff
@@ -1243,6 +1318,14 @@ static const struct uart_driver_api uart_ns16550_driver_api = {
 		irq_enable(DT_INST_IRQN(n));                                  \
 	}
 
+#define UART_NS16550_IRQ_CONFIG_ACPI1(n)                                                           \
+	static void irq_config_func##n(const struct device *dev)                                   \
+	{                                                                                          \
+	}
+
+#define UART_NS16550_IRQ_CONFIG_PCIE0(n)                                                           \
+	_CONCAT(UART_NS16550_IRQ_CONFIG_ACPI, DT_INST_ON_BUS(n, acpi))(n)
```

Comment:
What if there are no ACPI instances but purely PCIe instances?

Reply:
The design change takecare UART instances on both ACPI and PCI or DTS and ACPI at the same time. (For example: UART0 on ACPI node, UART1 on PCI or UART0 on DTS and UART1 on PCI. But doesn't support ACPI and DTS at the same time.